src/js_parser.zig

```diff
@@ -17529,7 +17529,26 @@ fn NewParser_(
                             else => break :try_record_hook,
                         };
                         if (!ReactRefresh.isHookName(original_name)) break :try_record_hook;
-                        p.handleReactRefreshHookCall(e_, original_name);
+                        if (p.options.features.react_fast_refresh) {
+                            p.handleReactRefreshHookCall(e_, original_name);
+                        } else {
+                            // if we're in this branch then it is a server
+                            // component, error if they try to use `useState`
+                            if (bun.strings.eqlComptime(original_name, "useState") and
+                                // this feels dumb, what's a better way to detect that it's not React?
+                                !bun.strings.includes(p.source.path.text, "node_modules/react"))
+                            {
+                                p.log.addError(
+                                    p.source,
+                                    expr.loc,
+                                    std.fmt.allocPrint(
+                                        p.allocator,
+                                        "\"useState\" is not available in a server component. If you need interactivity, consider converting part of this to a Client Component.",
```

Comment:
Maybe say add "use client" to the top of the file so they don't need to think "how do I convert to a client component?"